File: 0429-n-ary-tree-level-order-traversal/0429-n-ary-tree-level-order-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> ans = new ArrayList<>();
        if(root == null) return ans;
        
        Queue<Node> q = new LinkedList<>();
        q.offer(root);
        while(q.size() > 0){
            int siz = q.size();
            List<Integer> level = new ArrayList<>();
            
            while(siz-- > 0){
                Node f = q.poll(); //get front of queue
                level.add(f.val);
                //push all children of 'f' node
                for(Node nl : f.children){
                    q.offer(nl);
                }
            }
            ans.add(new ArrayList<>(level));
        }
        return ans;
    }
}

File: 0429-n-ary-tree-level-order-traversal/0429-n-ary-tree-level-order-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> ans;
        if(root == NULL) return ans;
        queue<Node*> q;
        q.push(root);
        while(!q.empty()){
            int siz = q.size();
            vector<int> level;
            while(siz--){
                auto f = q.front();
                q.pop();
                level.push_back(f->val);
                //only one change is this------
                for(auto nextlevel : f->children){ //children is vector of node's children
                    q.push(nextlevel);
                }
            }
            ans.push_back(level);
        }
        return ans;
    }
};